client/landing/stepper/hooks/use-site-copy.tsx

```diff
@@ -67,22 +79,28 @@ export const useSiteCopy = (
 		clearSignupDestinationCookie();
 		setPlanCartItem( { product_slug: plan?.product_slug as string } );
 
-		const marketplaceProducts = purchases
-			.filter( ( purchase ) =>
-				[ 'marketplace_plugin', 'marketplace_theme' ].includes( purchase.productType )
+		const marketplacePluginProducts = ( purchases || [] )
+			.filter(
+				( purchase ) =>
+					[ 'marketplace_plugin', 'marketplace_theme' ].includes( purchase.productType ) &&
+					purchase.siteId === site?.ID
+			)
+			.filter(
+				( purchase ) =>
+					purchase.productType === 'marketplace_plugin' && purchase.siteId === site?.ID
```

Comment:
@kozer, I caught this now. Do we need both filters? I think the second filter is "overwriting" the first one.
```diff
-			.filter(
-				( purchase ) =>
-					[ 'marketplace_plugin', 'marketplace_theme' ].includes( purchase.productType ) &&
-					purchase.siteId === site?.ID
-			)
			.filter(
				( purchase ) =>
					purchase.productType === 'marketplace_plugin' && purchase.siteId === site?.ID
```

Reply:
Actually, it's not, it is applied after the first one, but it's unnecessary of course.

Reply:
Since it's merged, what do you think it's better to tackle this?
1. Creating a quick fix
2. Tacke it to your branch, as you heavily changed the hook?

Reply:
Do you know which one is the correct one?
I think `[ 'marketplace_plugin', 'marketplace_theme' ]`  is what we need to use, can you confirm it?

Reply:
Oh no! You are right @sejas ! it's overwritten!

Reply:
Here is the new pr that fixes that https://github.com/Automattic/wp-calypso/pull/73154

Reply:
Thanks for the quick fix!